pyfar/io/io.py

```diff
@@ -490,6 +492,356 @@ def default_audio_subtype(format):
+        format definition.
+    expressions : list[str], optional
+        The expressions in COMSOL are the pysical property of the calculated
+        solution, e.g. total sound pressure in Pa. The expression depends on
+        the physics and have unit. Multiple expressions are possible in one
+        file. This parameter defines which parts of the COMSOL data is
+        returned as a pyfar audio object. By default, all expressions get
+        returned. An example can be like ``expressions=['pabe.Lp_t']``.
+        A list of all expressions can be obtained with
+        :py:func:`~pyfar.io.comsol_read_header`.
+    parameters : dict
+        Solutions or expressions can be evalueted for different parameters,
+        such as differnet angles or other parameter. This input dict contains
+        the parameters, which sould be extracted from the comsol file. An
+        example can be like
+        ``parameters={'theta': [0.0, 0.7854], 'phi': [0., 1.5708]}``.
+        A list of all expressions can be obtained with
+        :py:func:`~pyfar.io.comsol_read_header`. Default is None, then all
+        parameters are read.
+
+    Returns
+    -------
+    data : Signal or FrequencyData
+        Returns a Signal or FrequencyData object depending on the input data.
+        The output data always have the cshape (#nodes, #expressions,
+        #parameter1, #parameter2, ...).
+    coordinates : Coordinates, optional
+        A Coordinates Object based on the Input path. The coordinate system is
+        always cartesian. If the input dimension is lower than three, the
+        missing dimensions are set to zero. If there are no positions given,
+        this parameter will not be returned. The cshape of this object is
+        (#nodes,)
+
+    Raises
+    ------
+    SyntaxError
+        Raises SyntaxError if the file format is not supported.
+    ValueError
+        Raises ValueError if the Frequency or Time data are inconvenient in the
+        input file.
+
+    Examples
+    --------
+    >>> import pyfar as pf
+    >>> import os.path
+    >>> path = os.path.join(os.getcwd(), '..', '..', \\
+    >>>     'tests', 'test_io_data', 'pressure_parametric.csv')
+    >>> expressions, _, parameters, _, _ = pf.io.read_comsol_header(path)
+    >>> expressions
+    ['pabe.p_t']
+    >>> parameters
+    {'theta': [0.0, 0.7854, 1.5708, 2.3562, 3.1416],
+     'phi': [0.0, 1.5708, 3.1416, 4.7124]}
+    >>> parameters['theta'] = parameters['phi'][1:]
+    {'theta': [1.5708, 3.1416, 4.7124],
+     'phi': [0.0, 1.5708, 3.1416, 4.7124]}
+    >>> data, coordinates = pf.io.read_comsol(path, parameters=parameters)
+    >>> data
+    FrequencyData:
+    (8, 1, 3, 4) channels with 2 frequencies
+    >>> coordinates
+    1D Coordinates object with 8 points of cshape (8,)
+    domain: cart, convention: right, unit: met
+    coordinates: x in meters, y in meters, z in meters
+    Does not contain sampling weights
+    """
```

Comment:
An example on how to remove the nans and reshape the audio object in case of `Specific combinations` in a `Parametric Sweep` would be nice.